### PR TITLE
ENH: adding view type annotation

### PR DESCRIPTION
**Pull request by gregcaporaso**
Note that this just adds the annotations - we're not doing anything with the view types yet.

### CODE REVIEW

File: qiime/sdk/workflow.py [hunk 1]
```diff
@@ -155,6 +174,16 @@ def from_function(cls, function, inputs, outputs, name, doc):
             doc=doc, import_path=import_path, function_name=function_name,
             parameters=parameters, results=results)
         signature = Signature(name, inputs, outputs)
+
+        # just to show that we've got the views
+        input_views = {input: function.__annotations__[input]
+                       for input in inputs.keys()}
+        output_view_types = qiime.core.type.util.tuplize(
+                           function.__annotations__['return'])
+        output_views = dict(zip(outputs.keys(), output_view_types))
```
Comment (wasade) [inline qiime/sdk/workflow.py]: i dont think it is necessary to use `keys` here and on line 180

File: qiime/sdk/workflow.py [hunk 2]
```diff
@@ -155,6 +174,16 @@ def from_function(cls, function, inputs, outputs, name, doc):
             doc=doc, import_path=import_path, function_name=function_name,
             parameters=parameters, results=results)
         signature = Signature(name, inputs, outputs)
+
+        # just to show that we've got the views
+        input_views = {input: function.__annotations__[input]
+                       for input in inputs.keys()}
+        output_view_types = qiime.core.type.util.tuplize(
```
Comment (wasade) [inline qiime/sdk/workflow.py]: that's going a bit deep. would you be supportive of the creation of an issue oriented around import aliases?

**Comment by wasade** [comment]
Function annotations will open all new exciting doors for obfuscating code. Absolutely in support of using them, but just as an example of some of the interesting things that can be done:

```python
09:52:26 (daniel@sandbar):diversity> a = np.arange(5)

09:52:32 (daniel@sandbar):diversity> def foo() -> a:
    return "asd"
   ....:

09:52:37 (daniel@sandbar):diversity> foo()
Out[13]: 'asd'

09:52:39 (daniel@sandbar):diversity> foo.__annotations__
Out[14]: {'return': array([0, 1, 2, 3, 4])}

09:52:42 (daniel@sandbar):diversity> def bar() -> foo:
   ....:     return 123
   ....:

09:54:02 (daniel@sandbar):diversity> bar()
Out[16]: 123

09:54:03 (daniel@sandbar):diversity> bar.__annotations__
Out[17]: {'return': <function __main__.foo>}
```

It's a mix of awesome and scary.

**Comment by ebolyen** [comment]
@wasade Ideally, the annotations will be MyPy compatible, the things annotated will be a "view", so we could probably check to see if there even exists a registered view, throwing an error otherwise. (These are independent from the semantic types)

File: qiime/sdk/workflow.py (hunk 2 above)
Comment (ebolyen) [inline qiime/sdk/workflow.py]: I wouldn't be opposed :+1:

File: qiime/sdk/workflow.py (hunk 1 above)
Comment (gregcaporaso) [inline qiime/sdk/workflow.py]: fixed, thanks!

**Pull request merged**